types/buffer-layout/index.d.ts

```diff
@@ -0,0 +1,88 @@
+
```

Comment:
Required to satisfy any `buffer-layout` import since there is no publicly available types package. Anchor (and other repos) take a similar approach

https://github.com/coral-xyz/anchor/blob/master/ts/packages/borsh/types/buffer-layout/index.d.ts